### PR TITLE
Merge main to guardians 2023.05.26

### PR DESCRIPTION
**Pull request by iulianpascalau**

### CODE REVIEW

File: blockchain/proxy_test.go [hunk 1]
```diff
@@ -227,40 +228,40 @@ func TestGetAccount(t *testing.T) {
 	t.Run("nil address should error", func(t *testing.T) {
 		t.Parallel()
 
-		response, err := proxy.GetAccount(context.Background(), nil)
-		require.Equal(t, err, ErrNilAddress)
+		response, errGet := proxyInstance.GetAccount(context.Background(), nil)
+		require.Equal(t, errGet, ErrNilAddress)
```
Comment (sstanculeanu) [inline blockchain/proxy_test.go:232]:
```suggestion
		require.Equal(t, ErrNilAddress, errGet)
```

File: blockchain/proxy_test.go [hunk 2]
```diff
@@ -227,40 +228,40 @@ func TestGetAccount(t *testing.T) {
...
 		require.Nil(t, response)
 	})
 	t.Run("invalid address should error", func(t *testing.T) {
 		t.Parallel()
 
 		invalidAddress := data.NewAddressFromBytes([]byte("invalid address"))
-		response, err := proxy.GetAccount(context.Background(), invalidAddress)
-		require.Equal(t, err, ErrInvalidAddress)
+		response, errGet := proxyInstance.GetAccount(context.Background(), invalidAddress)
+		require.Equal(t, errGet, ErrInvalidAddress)
```
Comment (sstanculeanu) [inline blockchain/proxy_test.go:240]:
```suggestion
		require.Equal(t, ErrInvalidAddress, errGet)
```

File: cmd/cli/main.go [hunk 3]
```diff
@@ -428,33 +429,34 @@ func generateAndSendTransaction(options *selectedOptions, proxy interactors.Prox
 		return err
 	}
 
-	tx, err := ti.ApplyUserSignatureAndGenerateTx(options.senderCryptoHolder, options.txArguments)
+	copiedTx := options.tx
+	err = ti.ApplyUserSignature(options.senderCryptoHolder, &copiedTx)
 	if err != nil {
 		log.Error("error creating transaction", "error", err)
```
Comment (sstanculeanu) [inline cmd/cli/main.go:435]:
```suggestion
		log.Error("error signing transaction", "error", err)
```

File: cmd/cli/main.go [hunk 4]
```diff
@@ -463,23 +465,23 @@ func generateAndSendTransaction(options *selectedOptions, proxy interactors.Prox
...
+	err = sendFundWalletsTxs(td, proxy, tx, receivers)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func sendFundWalletsTxs(td *testData, proxy workflows.ProxyHandler, txArgs data.ArgCreateTransaction, receivers []core.AddressHandler) error {
+func sendFundWalletsTxs(td *testData, proxy workflows.ProxyHandler, txArgs transaction.FrontendTransaction, receivers []core.AddressHandler) error {
```
Comment (sstanculeanu) [inline cmd/cli/main.go:484]:
```suggestion
func sendFundWalletsTxs(td *testData, proxy workflows.ProxyHandler, tx transaction.FrontendTransaction, receivers []core.AddressHandler) error {
```

File: cmd/cli/main.go [hunk 5]
```diff
@@ -499,20 +501,21 @@ func sendFundWalletsTxs(td *testData, proxy workflows.ProxyHandler, txArgs data.
...
+	var tx transaction.FrontendTransaction
 	txArgs.Value = "10000000000000000000" // 10EGLD
 	for _, addressHandler := range receivers {
-		txArgs.RcvAddr = addressHandler.AddressAsBech32String()
-		fundingWalletCryptoHolder, err := cryptoProvider.NewCryptoComponentsHolder(keyGen, td.skFunding)
-		if err != nil {
+		txArgs.Receiver = addressHandler.AddressAsBech32String()
+		fundingWalletCryptoHolder, localErr := cryptoProvider.NewCryptoComponentsHolder(keyGen, td.skFunding)
+		if localErr != nil {
 			return err
```
Comment (sstanculeanu) [inline cmd/cli/main.go:510]:
```suggestion
			return localErr
```

File: cmd/cli/main.go [hunk 6]
```diff
@@ -499,20 +501,21 @@ func sendFundWalletsTxs(td *testData, proxy workflows.ProxyHandler, txArgs data.
...
 		}
-		tx, err = ti.ApplyUserSignatureAndGenerateTx(fundingWalletCryptoHolder, txArgs)
-		if err != nil {
-			log.Error("error creating transaction", "error", err)
-			return err
+		tx = txArgs // copy
+		localErr = ti.ApplyUserSignature(fundingWalletCryptoHolder, &tx)
+		if localErr != nil {
+			log.Error("error creating transaction", "error", localErr)
```
Comment (sstanculeanu) [inline cmd/cli/main.go:515]:
```suggestion
			log.Error("error signing transaction", "error", localErr)
```

File: blockchain/proxy_test.go (hunk 1 above)
Comment (iulianpascalau) [reply blockchain/proxy_test.go:232]: switched

File: cmd/cli/main.go (hunk 3 above)
Comment (iulianpascalau) [reply cmd/cli/main.go:435]: changed

File: cmd/cli/main.go (hunk 5 above)
Comment (iulianpascalau) [reply cmd/cli/main.go:510]: good catch, fixed

File: cmd/cli/main.go (hunk 6 above)
Comment (iulianpascalau) [reply cmd/cli/main.go:515]: fixed

File: cmd/cli/main.go [hunk 7]
```diff
@@ -109,8 +110,8 @@ VERSION:
 	log        = logger.GetOrCreate("mx-sdk-go/cmd/cli")
 )
 
-var HOME = os.Getenv("HOME")
-var pathGeneratedWallets = HOME + "/MultiversX/testnet/filegen/output/walletKey.pem"
+var homePath = os.Getenv("HOME")
+var pathGeneratedWallets = homePath + "/MultiversX/testnet/filegen/output/walletKey.pem"
```
Comment (raduchis) [inline cmd/cli/main.go:114]: will this work on OSX or windows with of.Getenv("HOME")? 
The path "MultiversX/testnet/filegen/output/walletKey.pem" is very specific and will this be generated always? 
Is there no way to generate a walletKey.pem in the current folder?

File: interactors/interface.go [hunk 8]
```diff
@@ -26,8 +26,8 @@ type TxBuilder interface {
 
 // GuardedTxBuilder defines the component able to build and sign a guarded transaction
 type GuardedTxBuilder interface {
-	ApplyUserSignatureAndGenerateTx(cryptoHolder core.CryptoComponentsHolder, arg data.ArgCreateTransaction) (*data.Transaction, error)
-	ApplyGuardianSignature(cryptoHolderGuardian core.CryptoComponentsHolder, tx *data.Transaction) error
+	ApplyUserSignature(cryptoHolder core.CryptoComponentsHolder, tx *transaction.FrontendTransaction) error
```
Comment (raduchis) [inline interactors/interface.go:29]: :+1:

File: cmd/cli/main.go (hunk 7 above)
Comment (raduchis) [reply cmd/cli/main.go:114]: This does not come from this PR actually...

**Review by raduchis — APPROVED** [review]

File: cmd/cli/main.go (hunk 7 above)
Comment (iulianpascalau) [reply cmd/cli/main.go:114]: This is specific to the local system testnet. That will almost certain run only on linux.

**Pull request merged**